### ISSUE TITLE
comment moderate vs blockade icon

### ISSUE THREAD

**Issue by ophian** [s9y/Serendipity#247]
The comment set-to-moderate and the comment info spamblock set-to-blockade "buttons" use the same thumb down/up icon "button". I know spamblock wordfilter has an moderate option too, but I would always set this to reject, which would IMHO be a better icon in the comment info too and mark this as something different.

**Comment by yellowled** [comment]
Went with a reject/checked combo. That should suffice, I guess.

**Comment by ophian** [comment]
@yellowled I think this is getting better. I am not always sure about it, since this is one of these cases, where the icon could (?maybe better or not?) show the action to happen **or** the state it is currently set. (Which also appears for the bayes hooked icons/actions.)

Btw, I fell over another icon in the dashboard comments collapsible preview/full open/close buttons since this box opens upwards. This could be an icon-up-dir instead of the icon-down-dir for the open box, which would make more sense (by keeping the icon-down-dir icon in the iconfont for other purposes).

And talking about this, my local repo and the netz blog differ on the backends comment page, since netzblog offers the collabsible box icon, showing the preview as a one-liner, my local repo does not have that at all and has the comment content opened (I have some 3/4-liner comment examples). How can this be? The git checkout and my local repo just differ in some other unrelated minor things.

**Comment by yellowled** [comment]
I'm not likely do change anything on the icon set any time soon, because

1. Other than from you, there has been zero feedback on the icon, which means the icon/title combo is working pretty well for most users.
2. Exchanging icons can be cumbersome.

Truth be told, there are icons in the backend that don't work well, but there are no good alternatives for these in FontAwesome. I picked FA because it offers by far the most interface icons of all the sets available on Fontello, but ideally, we would need a custom-made icon set for the backend. However, that's not likely to happen (icon design is hard, it's not something anyone can do).

> How can this be?

I don't know. The nbe dev blog needs to be updated, as of now, it's kind of a weird mixture of RC1 and some changes I deployed manually. Then again, you have *always* had issues with or differences in your local dev blog that no one else had, soooo …

**Comment by ophian** [comment]
> Then again, you have always had issues with or differences in your local dev blog that no one else had, soooo …

If you judge every comment by this, which by the way is not true, I'll stop contributing. Thanks!

**Comment by yellowled** [comment]
All I'm saying is it is kind of likely that it is an issue with your local setup which is next to impossible to reproduce for anyone but you.

I remember *multiple* instances where you reported something that I or anyone else was not able to reproduce. Might be related to you using Firefox, might be related to some Firefox addon you use, might be related to your local server. I don't know.

None of this was meant as an offense towards you.

**Comment by garvinhicking** [comment]
I'll check my local repo against nbe next week whether I see those diffs, too.

As for the icons I haven't checked out the current ones yet, will also report back on it. But I understand that changing them might also disturb current rc2 users eho might have gotten accustomed to the toggle items already. We should just be consistent for toggle arrow usage.

**Comment by yellowled** [comment]
> But I understand that changing them might also disturb current rc2 users eho might have gotten accustomed to the toggle items already.

That's actually **not** my biggest concern.

If someone were to offer us a custom-made iconset tomorrow, I'd take it in a heartbeat, no matter if changing icons would confuse RC users. That's not likely to happen, however.

**Comment by garvinhicking** [comment]
Hey,

ok so I checked on this and I also believe that we need a proper "UI" check on this.

See:
https://www.dropbox.com/s/c0xka2jj4icpjl7/Screenshot%202015-01-05%2011.44.48.png?dl=0
https://www.dropbox.com/s/4dsumh6kojg4qu4/Screenshot%202015-01-05%2011.44.59.png?dl=0

(BTW, also note the spacing mismatch of the spam-buttons?)

We do have a problem that usually in Serendipity we display the icon with an action that will happen when we click on it (I think). So for the category changer, it shows the icon of how the view will look like, when we click on it.

Now our "fold-in" and "fold-out" actually are not in line with that. When something is hidden, we have a cursor to the "right", however the proper position would be "bottom". Once we show something the cursor changes to "bottom", when actually it would need to show "up". Prevoisuly we had "+" and "-" buttons for that, where the indiciation was actually ok because it read "+" when something was folded and "-" when something was unfolded. So I do think we need to work on those toggle icons to make the UI clear. 

@yellowled Do you have an easy way to check the template for ambiguous use of toggle-icons? It would be great if we could make a check to see if every toggle always shows the icon as what WILL happen instead of what the current situation is like...?

@ophian In all my 3 local installations the comment-toggle functionality works. I assume you are testing with comments that are not long enough. You need to have a comment with more than 100 CHARACTERS. The amount of lines does not matter.

**Comment by yellowled** [comment]
> BTW, also note the spacing mismatch of the spam-buttons?

That's not how these buttons are supposed to look like.

https://www.dropbox.com/s/eb0cu7wgqz0l7zq/Screenshot%202015-01-05%2013.02.33.png?dl=0

I don't know how/why you get the misplaced buttons with the text visible. I don't have those in either the nbe dev blog (running latest git snapshot) or my blog (running RC2) in Chrome 39/OS X. Browser/template cache maybe?

> When something is hidden, we have a cursor to the "right", however the proper position would be "bottom".

Kind of a slight objection: these arrow icons are consistent with the browser widget for `<details>` (see for example Content/Categories). We will not be able to influence the way browsers render the `<details>` element.

> Do you have an easy way to check the template for ambiguous use of toggle-icons?

Other than going through the code by hand? No. (I will probably not have time for this over the next couple of days.)

**Comment by garvinhicking** [comment]
Hi!


> That's not how these buttons are supposed to look like.

I am not using the bayes plugin, my Links are from spamblock.

They look like this:

https://www.dropbox.com/s/9hfrlu1qp6ylxhi/Screenshot%202015-01-05%2013.39.48.png?dl=0

and they are outside of the <ul>. They get emitted by 
serendipity_event_spamblock, see "backend_view_comment" hooks. They only 
get shown when you use Akismet, I believe.

>     When something is hidden, we have a cursor to the "right", however
>     the proper position would be "bottom".
>
> Kind of a slight objection: these arrow icons are consistent with the
> browser widget for |<details>| (see for example Content/Categories). We
> will not be able to influence the way browsers render the |<details>|
> element.

Hm, I see. Actually that's kinda bad by browsers, or am I the only one 
who sees this as a UI issue? @onli?

**Comment by onli** [comment]
> Hm, I see. Actually that's kinda bad by browsers, or am I the only one who sees this as a UI issue? @onli?

That's one of those areas where it's almost not possible to predict what would work best. It depends on the model in the head, and to try to be consistent with the browser is probably the best we can do here. We'd have to test it with our users otherwise…

What I'd do to solve that is to try to find another solution to not tackle the problem directly, to get that function out of the button bar. A more link or something. Though a button is more sound here than a link… not too easy.

**Comment by yellowled** [comment]
> They only get shown when you use Akismet, I believe.

Ah, thanks for clearing that up. I don't use Akismet. I'll have a look at it (probably not in the next few days, though).

> Actually that's kinda bad by browsers, or am I the only one who sees this as a UI issue?

It's kind of pointless to discuss this. I have seen the arrow left/down elsewhere as well, for example it's pretty common in [accordion plugins](http://jqueryui.com/accordion/). It's an established UI pattern, whether one thinks it's logical or not. Pretty much the same with the hamburger icon.

**Comment by ophian** [comment]
> I assume you are testing with comments that are not long enough. You need to have a comment with more than 100 CHARACTERS. The amount of lines does not matter.

Well, **that** is an answer that clears things up! This is the case, yes! Thanks for the clarification.
...
I would be nice to have the summary happen as a 1-liner here too. How about using this 
`{$comment.fullBody|strip|replace:'<br />':''|truncate:120:"&hellip;"}`
and show the toggle button ever?
And/or by replacing linebreaks with some noticable (_the inline is just for playing around!_) replacement, eg
`{$comment.fullBody|strip|replace:'<br />':'<|>'|truncate:120:"&hellip;"|replace:'<|>':'<em style="color: red"> /</em>'}`

**Comment by garvinhicking** [comment]
@ophian Actually I'd like to have the newlines there, stripping them would make reading much harder...

About the UI thing: OK, I guess then we'll need to live with an established UI pattern. I was thinking about making the category selector in the popup a "radio button" like toggle, so that we display both buttons, and either one can be "highlighted" to indicate what view you are currently seeing. But that would mean a lot of work for a real small thing, and I think the current toggle is acceptable, too.

**Comment by yellowled** [comment]
> But that would mean a lot of work for a real small thing, and I think the current toggle is acceptable, too.

I'm also pretty sure almost anyone is going to click this button only once. :)

**Comment by yellowled** [comment]
Note: Please leave this issue open so I have it as a reminder to address the Akismet buttons.
